File: backend/src/contexts/backoffice/media/application/media_searcher.py
from src.contexts.backoffice.media.domain import Media, MediaRepository
from src.contexts.shared.domain.criteria import Criteria


class MediaSearcher:
    def __init__(self, repository: MediaRepository) -> None:
        self.repository = repository

    def run(self, criteria: dict | None = None) -> list[Media]:
        if criteria is None:
            return self.repository.search_all()
        return self.repository.matching(Criteria.from_primitives(**criteria))

File: frontend/src/contexts/backoffice/media/application/MediaSearcher.ts
import { Media } from "@/contexts/backoffice/media/domain/Media";
import { MediaRepository, PaginatedResponse } from "@/contexts/backoffice/media/domain/MediaRepository";

export class MediaSearcher {
    private readonly mediaRepository: MediaRepository;

    constructor(mediaRepository: MediaRepository) {
        this.mediaRepository = mediaRepository;
    }

    async run(): Promise<PaginatedResponse<Media>> {
        return await this.mediaRepository.matching();
    }
}
